### ISSUE TITLE
I can't create an arbitrary number of nested forms:

### ISSUE THREAD

**Issue by Qsppl** [hybridsjs/hybrids#248]
## This is what I'm trying to create:
"Project" creation form.

In the same form, you can write down as many “Contact Persons” of the target “Project” as you need.
We do not know in advance how many “Contact Persons” the project will have.

![image](https://github.com/hybridsjs/hybrids/assets/22717464/e4615a4c-e76b-456d-87ab-6f0fdd7f6128)

## Problem:
I cannot implement the functionality of recording an _arbitrary number_ of Contact Persons in the form.

I can't create an arbitrary number of drafts from one model definition, like a `store([ContactStore], { draft: true })`. 

Also I can't create/delete a separate draft via "store direct methods", like a `host.contactDrafts.push(store.draft(ContactStore))`.

Also, I can't create a structure from drafts as from models, like a `host.projectDraft.contacts.push(store.draft(ContactStore))`.

## Here is a sandbox with a ready-made example of a form:
![image](https://github.com/hybridsjs/hybrids/assets/22717464/fd61d4e3-a219-416c-861d-05c1b5c13fb8)

**Comment by smalluban** [comment]
You can ;) The below example uses two models, each has its own component for the form. You don't have to submit `Person` drafts, just when the whole form is submitted. Drafts when disconnected from DOM will be cleared.

https://stackblitz.com/edit/hybrids-simple-counter-vqdzk3?file=src%2Findex.js

**Comment by Qsppl** [comment]
Thanks a lot, it works.

**Comment by Qsppl** [comment]
However, I cannot use `store.set(Definition, draft)` instead of `store.submit(draft)` because this approach causes problems when _updating_ the model.

**Comment by smalluban** [comment]
If the model references another model, the draft of that model still references the second one (not its draft).

The draft mode is nothing more like making a copy of the original and making it memory accessible (if the instance is not there). The reference does not create another deep draft. This is a core issue here. 

From the store perspective, Project and Person models are separate. It means, that if you want to reference a Person in a Project, you must "save" it first. I know, that what you would like to do, is to "create" it together - project and all of the persons (I assume that your API allows you to do so). 

However, it would be more complicated if you edit the existing Project, as it references Persons, which when set will update accordingly. If this is your requirement, you should create a "special" model for this form - this is not that bad - you can spread fields from both original models, create nested persons, and after saving it, clear original models if needed (just not spread the `[store.connect]` - so both models will be memory-based copies). Then if you need to edit the model you can get the original model with included persons.

To sum up, the store is flexible and it does not always reflect the API data - it should rather reflect the structure needed in the view. In short, my suggestion is just to make a custom "draft" mode - there is not that much magic behind the feature.

**Comment by Qsppl** [comment]
We accomplished this task by creating a `PersonDraft` model, but it looks bad. I'll try to write my own implementation of `store.draft()`
